packages/node_modules/@webex/plugin-meetings/src/share.js

```diff
@@ -0,0 +1,110 @@
+/*!
+ * Copyright (c) 2015-2018 Cisco Systems, Inc. See LICENSE file.
+ */
+
+export const TYPES = {
+  application: 'application',
+  screen: 'screen',
+  stream: 'stream',
+  whiteboard: 'whiteboard'
+};
+
+export const STATUSES = {
+  idle: 'idle',
+  ongoing: 'ongoing',
+  terminating: 'terminating',
+  share_failed: 'share_failed',
+  stop_share_failed: 'stop_share_failed',
+  starting: 'starting'
+};
+
+/**
+ * Events to emit
+ * ==============
+ *
+ * - change (object):
+ *   - status (string): See STATUSES.
+ *   - reason (string)
+ */
+export default class Share {
+  /**
+   * Create a Share.
+   *
+   * TODO: It may be more appropriate for this class
+   * to be comprised of helpful static methods.
+   * @param {object} options
+   * @param {object} options.resolution
+   * @param {object} options.framerate
+   * @param {string} options.type
+   */
+  constructor({resolution, framerate, type}) {
+    this.resolution = resolution;
+    this.framerate = framerate;
+    this.status = STATUSES.idle;
+    if (Object.values(TYPES).includes(type)) {
+      this.type = type;
+    }
+    else {
+      throw new Error(`Must be one of the following types: ${TYPES}`);
```

Comment:
May want to do `Object.values(TYPES)` here like in line 44 to give a little more friendly error message.